lib/consumer.ex

```diff
@@ -284,7 +284,7 @@ defmodule GenRMQ.Consumer do
   `message` - `GenRMQ.Message` struct
   """
   @spec ack(message :: %GenRMQ.Message{}) :: :ok
-  def ack(%Message{state: %{in: channel}, attributes: %{delivery_tag: tag}} = message) do
```

Comment:
We could consider keeping state propety in the message, limited to `%{in: channel}`. It would be at least smaller breaking change.